### PR TITLE
Add Brotli support to SocketsHttpHandler

### PR DESCRIPTION
**Pull request by stephentoub**
- Add DecompressionMethods.Brotli
- Add support for Brotli to SocketsHttpHandler
- Add some tests

Fixes https://github.com/dotnet/corefx/issues/26995
cc: @ianhays, @joshfree, @geoffkizer, @davidsh

### CODE REVIEW

**Comment by joshfree** [comment]
Thanks @stephentoub 

related asp.net issue https://github.com/aspnet/BasicMiddleware/issues/217

File: src/System.Net.Primitives/src/System/Net/DecompressionMethods.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ public enum DecompressionMethods
     {
         None = 0,
         GZip = 1,
-        Deflate = 2
+        Deflate = 2,
```
Comment (geoffkizer) [inline src/System.Net.Primitives/src/System/Net/DecompressionMethods.cs]: Nit: Why not specify using binary (or hex) constants instead of decimal?

Comment (stephentoub) [reply src/System.Net.Primitives/src/System/Net/DecompressionMethods.cs]: I was just following suit on what was already done.  But I can change it to hex.

**Comment by stephentoub** [comment]
@dotnet-bot test Windows x64 Debug Build please (lots of unrelated failures on the Nano server leg... looks like the system is in a bad state)

**Pull request merged**